common/app/assets/assets.scala

```diff
@@ -120,27 +106,33 @@ object css {
 }
 
 object js {
-  val curl: String = Get(LoadFromClasspath("assets/curl.js").map(RelativePathEscaper.escapeLeadingDotPaths))
-  val polyfillioUrl: String = Get(LoadFromClasspath("assets/polyfill.io").map(RelativePathEscaper.escapeLeadingDotPaths)).trim
+  val curl: String = RelativePathEscaper.escapeLeadingDotPaths(AssetLoader.loadFromClasspath("assets/curl.js"))
+  val polyfillioUrl: String = RelativePathEscaper.escapeLeadingDotPaths(AssetLoader.loadFromClasspath("assets/polyfill.io")).trim
 }
 
-object Get {
-  def apply[T](`try`: Try[T]) = `try` match {
-    case Success(s) => s
-    case Failure(e) => throw e
+case class AssetNotFoundException(assetPath: String) extends Exception(s"Cannot find asset $assetPath. You should run `make compile`.")
+
+object AssetLoader {
+
+  private val memoizedAssets: ConcurrentMap[String, String] = TrieMap()
+
+  def load(assetPath: String)(implicit context: ApplicationContext): String = {
+    if (context.environment.mode == Mode.Dev) {
+      loadFromClasspath(assetPath)
+    } else {
+      loadMemoized(assetPath)
+    }
   }
-}
 
-// gets the asset url from the classpath
-object LoadFromClasspath {
-  def apply(assetPath: String): Try[String] = {
-    (Option(this.getClass.getClassLoader.getResource(assetPath)) match {
-      case Some(s) => Success(s)
-      case None => Failure(AssetNotFoundException(assetPath))
-    }).flatMap { url =>
-      Try(IOUtils.toString(url))
+  def loadMemoized(assetPath: String): String = memoizedAssets.getOrElseUpdate(assetPath, loadFromClasspath(assetPath))
+
+  def loadFromClasspath(assetPath: String): String = {
+    val assetUrl: Option[URL]  = Option(this.getClass.getClassLoader.getResource(assetPath))
+    assetUrl.map(url => Try(IOUtils.toString(url))) match {
+      case Some(Success(s)) => s
+      case Some(Failure(e)) => throw e
```

Comment:
Ideally yes. This `Asset` file needs a proper refactoring. I didn't want to change its behaviour while still being able to make it reusable. 
I agree that if we decide to go with this solution I will spend some times rethinking the existing error handling.